lightkurve/seismology/core.py

```diff
@@ -276,11 +353,170 @@ def plot_echelle(self, deltanu=None, numax=None,
+                         smooth_filter_width=.1,
+                         scale='linear', plot_width=490, plot_height=340, title='Echelle'):
+        """Helper function to make the elements of the echelle diagram for bokeh plotting.
+        """
+
+        if not hasattr(deltanu, 'unit'):
+            deltanu = deltanu * self.periodogram.frequency.unit
+
+        if smooth_filter_width:
+            pgsmooth = self.periodogram.smooth(filter_width=smooth_filter_width)
+            freq = pgsmooth.frequency  # Makes code below more readable below
+            power = pgsmooth.power     # Makes code below more readable below
+        else:
+            freq = self.periodogram.frequency  # Makes code below more readable
+            power = self.periodogram.power     # Makes code below more readable
+
+
+        ep, x_f, y_f = self._clean_echelle(deltanu=deltanu,
+                         minimum_frequency=minimum_frequency, maximum_frequency=maximum_frequency,
+                         smooth_filter_width=smooth_filter_width,
+                         scale=scale)
+
+        fig = figure(plot_width=plot_width, plot_height=plot_height,
+                     x_range=(0, 1), y_range=(y_f[0].value, y_f[-1].value),
+                     title=title, tools='pan,box_zoom,reset',
+                     toolbar_location="above",
+                     border_fill_color="white")
+
+        fig.yaxis.axis_label = r'Frequency [{}]'.format(freq.unit.to_string())
+        fig.xaxis.axis_label = r'Frequency / {:.3f} Mod. 1'.format(deltanu)
+
+        lo, hi = np.nanpercentile(ep.value, [0.1, 99.9])
+        vlo, vhi = 0.3 * lo, 1.7 * hi
+        vstep = (lo - hi)/500
+        color_mapper = LogColorMapper(palette="RdYlGn10", low=lo, high=hi)
+
+        fig.image(image=[ep.value], x=0, y=y_f[0].value,
+                  dw=1, dh=y_f[-1].value,
+                  color_mapper=color_mapper, name='img')
+
+        stretch_slider = RangeSlider(start=vlo,
+                                     end=vhi,
+                                     step=vstep,
+                                     title='',
+                                     value=(lo, hi),
+                                     orientation='vertical',
+                                     width=10,
+                                     height=230,
+                                     direction='rtl',
+                                     show_value=False,
+                                     sizing_mode='fixed',
+                                     name='stretch')
+
+        def stretch_change_callback(attr, old, new):
+            """TPF stretch slider callback."""
+            fig.select('img')[0].glyph.color_mapper.high = new[1]
+            fig.select('img')[0].glyph.color_mapper.low = new[0]
+
+        stretch_slider.on_change('value', stretch_change_callback)
+        return fig, stretch_slider
+
+
+    def interact_echelle(self, notebook_url='localhost:8888', **kwargs):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Seismology.interact_echelle is too complex (16)](https://app.codacy.com/app/barentsen/lightkurve/pullRequest?prid=4468966)